busy_indicator_view.py

```diff
@@ -56,23 +56,18 @@ def on_activated_async(self, active_view):
         draw(**State)
 
 
-indicators = [
-    'Linting.  ',
-    'Linting.. ',
-    'Linting. .',
-    'Linting ..',
-    'Linting  .',
-]
+def tick():
+    while True:
+        for phase in ('.  ', '.. ', '. .' ' ..', '  .'):
+            yield "Linting" + phase
 
 
 def draw(active_view, running, **kwargs):
     buffer_id = active_view.buffer_id()
     start_time = running.get(buffer_id, None)
     now = time.time()
     if start_time and (INITIAL_DELAY <= (now - start_time) < TIMEOUT):
-        num = len(indicators)
-        text = indicators[int((now - start_time) * 1000 / CYCLE_TIME) % num]
-        active_view.set_status(STATUS_BUSY_KEY, text)
+        active_view.set_status(STATUS_BUSY_KEY, next(tick()))
         sublime.set_timeout_async(lambda: draw(**State), CYCLE_TIME)
```

Comment:
If you do animations, probably in JS or so, you always reference a timer. `now` `then` etc.

Bc you actually never know when you're callback runs. So you actually don't need a cycler that gives you a new frame per call. You have to write a function that gives you the correct frame at given time.